modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/MultiEncodeWorkflowOperationHandler.java

```diff
@@ -0,0 +1,545 @@
+    if (sourceTags == null && sourceFlavors == null)
+      throw new WorkflowOperationException("No source tags or Flavor");
+    if (profiles == null)
+      throw new WorkflowOperationException("Missing profiles");
+    if (sourceTags != null) { // If source tags are used to select tracks
+      // If use source and target tags, there should be the same number of them or all map into one target
+      if (targetTags != null && (targetTags.length != 1 && sourceTags.length != targetTags.length))
+        throw new WorkflowOperationException("number of source tags " + sourceTags.length
+                + " does not match number of target tags " + targetTags.length + " (must be the same or one target)");
+      // There should be the same number of source tags or profile groups or all use same group of profiles
+      if (profiles.length != 1 && sourceTags.length != profiles.length) {
+        throw new WorkflowOperationException(
+                "number of source tags segments " + sourceTags.length + " does not match number of profiles segments "
+                        + profiles.length + " (must be the same or one profile)");
+      }
+      // If use source tags and source flavors, there should be the same number of them or one
+      if (sourceFlavors != null && (sourceTags.length != 1 && sourceFlavors.length != 1)
+              && sourceFlavors.length != sourceTags.length) {
+        throw new WorkflowOperationException("number of source tags segments " + sourceTags.length
+                + " does not match number of source Flavor segments " + sourceFlavors.length
+                + " (must be the same or one)");
+      }
+      n = sourceTags.length; // at least this many tracks
+    }
+    if (sourceFlavors != null) { // If flavors are used to select tracks
+      // If use source and target flavors, there should be the same number of them or all map into one target
+      if (targetFlavors != null && (targetFlavors.length != 1 && sourceFlavors.length != targetFlavors.length)) {
+        throw new WorkflowOperationException(
+                "number of source flavors " + sourceFlavors.length + " segment does not match number of target flavors"
+                        + targetFlavors.length + " (must be the same or one target flavor)");
+      }
+      // If use target tags, there should be the same number of source flavors and target tags or all map into one
+      // target tag
+      if (targetTags != null && (targetTags.length != 1 && sourceFlavors.length != targetTags.length)) {
+        throw new WorkflowOperationException(
+                "number of source flavors " + sourceFlavors.length + " segment does not match number of target Tags"
+                        + targetTags.length + " (must be the same or one target)");
+      }
+      // Number of profile groups should match number of source flavors
+      if ((profiles.length != 1 && sourceFlavors.length != profiles.length)) {
```

Comment:
Similar to a comment above: If you define the selected tracks to be the intersection of tracks located by both source-flavors and source-tags, it would be valid to have profiles.length source-tags but just one source-flavor